sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -641,4 +641,30 @@ object StructType extends AbstractDataType {
     fields.foreach(s => map.put(s.name, s))
     map
   }
+
+  /**
+   * Returns a `StructType` that contains missing fields recursively from `source` to `target`.
+   * Note that this doesn't support looking into array type and map type recursively.
```

Comment:
Where does this limitation come?; we don't need to support this case, or supporting it is technically difficult?

Reply:
I leverage `WithFields` to add missing nested fields into structs. `WithFields` doesn't support array or map types currently.